qucs/qucs-filter/qucsfilter.cpp

```diff
@@ -666,57 +675,82 @@ void QucsFilter::FlushImage()
-  {
-    MicrostripcheckBox->setEnabled(false);
-    MicrostripcheckBox->setChecked(false);
-    ComboClass->setEnabled(true);
-  }  
-  else
+
+  if (index < 2)//LC filters
   {
-   MicrostripcheckBox->setEnabled(true);
-  if(index <= 3) {  // set to "transmission line" types?
-    // set to bandpass fixed
-    ComboClass->setCurrentIndex(CLASS_BANDPASS);
-    slotClassChanged(CLASS_BANDPASS);
-    ComboClass->setEnabled(false);
-  }
-  else if(index < 5) {  // set to "stepped impedance" types?
-    // set to lowpass fixed
-    ComboClass->setCurrentIndex(CLASS_LOWPASS);
-    slotClassChanged(CLASS_LOWPASS);
-    ComboClass->setEnabled(false);
-  }
-  else
-    ComboClass->setEnabled(true);
+     ComboType->clear();
+     ComboType->addItem("Bessel");
+     ComboType->addItem("Butterworth");
+     ComboType->addItem("Chebyshev");
+     ComboType->addItem("Cauer");
+     MicrostripcheckBox->setEnabled(false);
+     MicrostripcheckBox->setChecked(false);
+     box2->setEnabled(false);//Microstrip substrate definition panel
+     ComboClass->setEnabled(true);
+     return;
   }
-  if (index == 5)//Quarter wave coupled resonator
+
+  //Remove Cauer filters from ComboClass since they cannot be implemented with the other topologies
+  //Otherwise, the user may select it and face a number of annoying warnings...
+  ComboType->clear();
+  ComboType->addItem("Bessel");
+  ComboType->addItem("Butterworth");
+  ComboType->addItem("Chebyshev");
+
+
+  //Shared settings for microwave BP filters
+  if ((index == 2) || (index == 3) || (index == 5) || (index == 6))
   {
-     ComboClass->setCurrentIndex(CLASS_BANDPASS);
-     slotClassChanged(CLASS_BANDPASS);
+           ComboClass->setCurrentIndex(CLASS_BANDPASS);
+           slotClassChanged(CLASS_BANDPASS);
+           ComboClass->setEnabled(false);
+           MicrostripcheckBox->setEnabled(true);
+           MicrostripcheckBox->setChecked(false);//By default, the microstrip implementation is unchecked
+           box2->setEnabled(false);//Microstrip substrate definition panel
   }
-  if (index == 6)//Quarter wave ring
+  
+  if (index == 4)//Stepped impedance LPF
   {
+     ComboClass->setCurrentIndex(CLASS_LOWPASS);
+     slotClassChanged(CLASS_LOWPASS);
      ComboClass->setEnabled(false);
-     ComboType->setEnabled(false);
-     EditOrder->setEnabled(false);
+     MicrostripcheckBox->setEnabled(true);
+     MicrostripcheckBox->setChecked(false);//By default, the microstrip implementation is unchecked
+     box2->setEnabled(false);//Microstrip substrate definition panel
+  }
+
+  if (index == 6)//Quarter wavelength side coupled ring resonator (BP)
+  {
+     ComboType->setEnabled(false);//Conventional frequency responses not available
+     EditOrder->setEnabled(false);//Not possible to modify the order
      LabelStart->setText("1st transmission zero frequency:");
-     LabelStop->setText("Central frequency:");
+     LabelStop->setText("2nd transmission zero frequency::");
   }
   else
   {
```

Comment:
This block of code may be never reached at some conditions, because there is an another `return` around line 690.